web/src/app/admin/admin-message-logs/DebugMessagesList.tsx

```diff
@@ -0,0 +1,122 @@
+import React, { useEffect } from 'react'
+import { Box } from '@mui/system'
+import { DateTime } from 'luxon'
+import { DebugMessage } from '../../../schema'
+import DebugMessageCard from './DebugMessageCard'
+import { useFuse } from './useFuse'
+import { useURLParam } from '../../actions'
+import { Typography, Button } from '@mui/material'
+
+interface KeyedDebugMessage extends DebugMessage {
+  additonalKeys?: {
+    filteredDestination: string
+  }
+}
+
+interface Props {
+  debugMessages?: KeyedDebugMessage[]
+  selectedLog: DebugMessage | null
+  onSelect: (debugMessage: DebugMessage) => void
+  onLoadMore: () => void
+  onResetLoadMore: () => void
+  numRendered: number
+}
+
+export default function DebugMessagesList(props: Props): JSX.Element {
+  const {
+    debugMessages = [],
+    selectedLog,
+    onSelect,
+    numRendered,
+    onLoadMore,
+    onResetLoadMore,
+  } = props
+
+  const [searchTerm] = useURLParam('search', '')
+  const [start] = useURLParam('start', '')
+  const [end] = useURLParam('end', '')
+
+  const { setSearch, results } = useFuse<KeyedDebugMessage>({
+    data: debugMessages,
+    keys: [
+      'destination',
+      'userName',
+      'serviceName',
+      'status',
+      'additionalKeys.filteredDestination',
+    ],
+    options: {
+      shouldSort: false,
+      showResultsWhenNoSearchTerm: true,
+      ignoreLocation: true,
+      useExtendedSearch: true,
+    },
+  })
+
+  const startDT = start ? DateTime.fromISO(start) : null
+  const endDT = end ? DateTime.fromISO(end) : null
+
+  let filteredResults = results.slice() // copy results array
+  filteredResults = filteredResults.filter((result) => {
+    const createdAtDT = DateTime.fromISO(result.item.createdAt)
+    if (startDT && startDT > createdAtDT) return false
+    if (endDT && endDT < createdAtDT) return false
+    return true
+  })
+
+  // reset page load amount when filters change
+  useEffect(() => {
+    onResetLoadMore()
+  }, [searchTerm, start, end])
```

Comment:
This seems off, this component is triggering an on-change for fields it does not control (i.e., start and end are never changed here) also, it's being called on-render instead of on the change event. It makes it hard to follow the control flow, causes multiple renders for a single change, and can become a loop if `onResetLoadMore` changes any values.

We should change this to only be called on a change event. In this case though, there are no events being triggered here, only passed in, and `useFuse` should be updated to accept the search term instead of needing to be kept in-sync with calls to `setSearch`